src/main/java/org/mamute/dao/QuestionDAO.java

```diff
@@ -152,17 +153,15 @@ public Question load(Question question) {
 				.list();
 	}
 	
-	public List<Question> top(String section, int count, DateTime since) {
+	public List<Question> top(Tab tab, int count, DateTime since) {
 		Order order;
-		if (section.equals("viewed")) {
-			order = Order.desc("q.views");
-		}
-		else if (section.equals("answered")) {
-			order = Order.desc("q.answerCount");
-		}
-		else /*if (section.equals("voted"))*/ {
-			order = Order.desc("q.voteCount");
+
+		switch (tab.getType()) {
+			case VIEWED:    order = Order.desc("q.views");          break;
+			case ANSWERED:  order = Order.desc("q.answerCount");    break;
+			default:        order = Order.desc("q.voteCount");      break;
 		}
```

Comment:
You could create a method at the `Tab` enum that returns the Order for that tab.

Order order = tab.getOrder();